Modules/_testcapimodule.c

```diff
@@ -646,6 +645,42 @@ test_get_type_qualname(PyObject *self, PyObject *Py_UNUSED(ignored))
     Py_RETURN_NONE;
 }
 
+static PyObject *
+test_get_type_fullyqualname(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    PyObject *name = PyType_GetFullyQualifiedName(&PyLong_Type);
+    assert(PyUnicode_EqualToUTF8(name, "int"));
+    Py_DECREF(name);
+
+    name = PyType_GetFullyQualifiedName(&PyODict_Type);
+    assert(PyUnicode_EqualToUTF8(name, "collections.OrderedDict"));
+    Py_DECREF(name);
+
+    PyObject *HeapTypeNameType = PyType_FromSpec(&HeapTypeNameType_Spec);
+    if (HeapTypeNameType == NULL) {
+        Py_RETURN_NONE;
```

Comment:
Shouldn't this return NULL? You're returning with an exception set.

Reply:
Oh right, fixed. I copied just copied this code from test_get_type_qualname(). I fixed the 3 functions.